Training/Training/intlist.c

```diff
@@ -0,0 +1,188 @@
+#include "intlist.h"
+#include <malloc.h>
+#include<stdbool.h>
+
+struct LinkedList* Create () {
+   struct LinkedList* list = NULL;
+   do {
+      list = (struct LinkedList*)malloc (sizeof (struct LinkedList));
+      if (list != NULL) {
+         list->head = NULL;
+         list->isDeleted = false;
+         list->count = 0;
+      }
+   } while (list == NULL);
+   return list;
+}
+
+struct Node* CreateNode (int data) {
+   struct Node* newNode = NULL;
+   do {
+      newNode = (struct Node*)malloc (sizeof (struct Node));
+      if (newNode != NULL) {
+         newNode->data = data;
+         newNode->next = NULL;
+      }
+   } while (newNode == NULL);
+   return newNode;
+}
+
+int Delete (struct LinkedList* list) {
+   if (list->isDeleted == true)
+      return E_LIST_DNE;
+   if (list->head == NULL) {
+      list->isDeleted = true;
+      return 0;
+   }
+   struct Node* temp = NULL;
+   while (list->head != NULL) {
+      temp = list->head;
+      list->head = (list->head)->next;
+      free (temp);
+   }
+   list->isDeleted = true;
+   return 0;
+}
+
+int Add (struct LinkedList* list, int data) {
+   if (list->isDeleted == true)
+      return E_LIST_DNE;
+   struct Node* newNode = CreateNode (data);
+   // in case this is the first node, make the newNode as
+   // the head of the LinkedList
+   if (list->head == NULL) {
+      list->head = newNode;
+      list->count++;
+      return 0;
+   }
+   // Create a pointer to iterate till the last node
+   struct Node* current = list->head;
+   while (current->next != NULL)
+      current = current->next;
+   // make the next of the tail to the newNode
+   current->next = newNode;
+   list->count++;
+   return 0;
+}
+
+int Insert (struct LinkedList* list, int ind, int data) {
+   if (list->isDeleted == true)
+      return E_LIST_DNE;
+   struct Node* newNode = CreateNode (data);
+   // If inserting at the beginning
+   if (ind == 0) {
+      newNode->next = list->head;
+      list->head = newNode;
+      list->count++;
+      return 0;
+   }
+   int count = Count (list);
+   struct Node* current = list->head;
+   if (current == NULL) {
+      free (newNode);
+      return E_INVALID_INDEX;
+   }
+   if (ind >= 0 && ind < count) {
+      for (int i = 0; i < ind - 1; i++)
+         current = current->next;
+   } else { // If the position is out of bounds
+      free (current);
+      return E_INVALID_INDEX;
+   }
+   newNode->next = current->next;
+   current->next = newNode;
+   list->count++;
+   return 0;
+}
+
+int RemoveAt (struct LinkedList* list, int ind) {

```

Comment:
Use a more meaningful name like "index" instead of "ind". Nothing is gained by saving 2 characters.